packages/react-native/Libraries/StyleSheet/processTransformOrigin.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+import invariant from 'invariant';
+
+const INDEX_X = 0;
+const INDEX_Y = 1;
+const INDEX_Z = 2;
+
+/* eslint-disable no-labels */
+export default function processTransformOrigin(
+  transformOrigin: Array<string | number> | string,
+): Array<string | number> {
+  if (typeof transformOrigin === 'string') {
+    const transformOriginString = transformOrigin;
+    const regex = /(top|bottom|left|right|center|\d+(?:%|px)|0)/gi;
+    const transformOriginArray: Array<string | number> = ['50%', '50%', 0];
+
+    let index = INDEX_X;
+    let matches;
+    outer: while ((matches = regex.exec(transformOriginString))) {
+      let nextIndex = index + 1;
+
+      const value = matches[0];
+      const valueLower = value.toLowerCase();
+
+      switch (valueLower) {
+        case 'left':
+        case 'right': {
+          invariant(
```

Comment:
It's subjective, but there was some discussion within the team before on whether we should be permissive of invalid styles during development. E.g. add a warning, instead of throwing an exception. I think we changed some other to be more permissive and just warn and no-op, since unhandled exception is disruptive during development/live reload, but I could see both ways.

Reply:
cc - @jacobp100 need to handle this in all PRs. wdyt? maybe use console.warn?

Reply:
Yeah I can make changes to this. If you let me know what way you want to go I'll update it in my PR (#38626)